### DISCUSSION TITLE
Using Hangfire in internal command processing

### DISCUSSION THREAD

**Discussion by mehdihadeli** [kgrzybek/modular-monolith-with-ddd#187] [Ideas]
Hi @kgrzybek,
Instead of handling internal commands processing and its persistence mechanism for retrying if they failed, is not better using hangfire for command executing like your article? 
[Processing commands with Hangfire and MediatR](http://www.kamilgrzybek.com/design/processing-commands-with-hangfire-and-mediatr/)

**Comment by kgrzybek** [comment]
Hi @mehdihadeli ,

Yes, you can use Hangfire for this kind of stuff. I used Hangfire exactly for that purpose in one of my commercial projects and it worked great.

I even consider usage of that library in one of the module instead of Quartz to show how it can be used.

**Comment by mehdihadeli** [comment]
@kgrzybek Thanks for your answer,

for using hangfire which approach is better?

1. Enqueue command to hangfire (with hangfire Enqueue method) with considering some internal retry feature of hangfire (for example 3)
2. Like your implemented approach in your modules we insert our enqueue command to module's `InternalCommand` table then using hangfire recurring job for reading from this table and if `ProcessedDate` is `Null` dispatch this command and if it was success update this `ProcessedDate` to current time

I in first approach if after 3 retries(hangfire retry=3) with hangfire internal mechanism and fail finally, we lost our command to a successful execution.
What's your opinion?
